### PR TITLE
[CI] stop e2e test prebuild

### PR DESCRIPTION
**Pull request by sausagee**
## Motivation
smoketest prebuilds libra-node inside libra-swarm. This breaks the
original assumption that CI can prebuild the test binary and run the
tests in parallel. It is possible to prebuild the libra-node binary,
but it requires saving and restoring the build artifacts in target/
between jobs, which takes longer than building from scratch.

## Test Plan
CI
image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -264,16 +245,18 @@ jobs:
       MESSAGE_PAYLOAD_FILE: "/tmp/message_payload"
     steps:
       - build_setup
-      - restore_cargo_package_cache
       - attach_workspace:
           at: /home/circleci/project
+      - restore_cargo_package_cache
```
Comment (rexhoffman) [inline .circleci/config.yml:250]: Got it, this is save since the cargo cache is placed in /usr/local/cargo, but this was an unnecessary change.  Puzzled on it for a bit.

File: .circleci/config.yml [hunk 2]
```diff
@@ -264,16 +245,18 @@ jobs:
...
       - run:
           name: Determine test targets for this container.
           # NOTE Currently the tests are distributed by name order. Once test
           # metadata is enabled, the tests can be distributed by run time to
           # speed up this job.
           command: |
-            ls -all target/debug/*
+            RUST_BACKTRACE=1 cargo x test --package testsuite -- --list | \
```
Comment (rexhoffman) [inline .circleci/config.yml:257]: Should be safe since it's not executing developer code in generation of the list, and so far our build is deterministic.

File: .circleci/config.yml (hunk 1 above)
Comment (sausagee) [reply .circleci/config.yml:250]: This restores the crate cache, not saving it. It is needed to speed up the build by avoiding fetching over network.

The two lines (`attach_workspace blah blah`) right above this are actually not needed anymore, as there is no more prebuilt binary to restore from a prev job.

Comment (sausagee) [reply .circleci/config.yml:250]: Fixed in #4016
